src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/ApplicationServices/SingleInstanceHelpers.vb

```
@@ -0,0 +1,106 @@
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Option Strict On
+Option Explicit On
+
+Imports System
+Imports System.IO
+Imports System.IO.Pipes
+Imports System.Runtime.Serialization
+Imports System.Threading
+
+Namespace Microsoft.VisualBasic.ApplicationServices
+
+    Friend Module SingleInstanceHelpers
+        Private Const NamedPipeOptions As PipeOptions = PipeOptions.Asynchronous Or PipeOptions.CurrentUserOnly
+
+        Friend Function CreatePipeServer(pipeName As String) As NamedPipeServerStream
+            Try
+                Return New NamedPipeServerStream(
+                        pipeName:=pipeName,
+                        direction:=PipeDirection.In,
+                        maxNumberOfServerInstances:=1,
+                        transmissionMode:=PipeTransmissionMode.Byte,
+                        options:=NamedPipeOptions)
+            Catch ex As IOException
+                Return Nothing
+            End Try
+        End Function
+
+        Friend Async Function WaitForClientConnectionsAsync(pipeServer As NamedPipeServerStream, callback As Action(Of String()), cancellationToken As CancellationToken) As Task
+            Try
+                While True
+                    cancellationToken.ThrowIfCancellationRequested()
+                    Await pipeServer.WaitForConnectionAsync(cancellationToken).ConfigureAwait(False)
+                    Try
+                        Dim args = Await ReadArgsAsync(pipeServer, cancellationToken).ConfigureAwait(False)
+                        If args IsNot Nothing Then
+                            callback(args)
+                        End If
+                    Finally
+                        pipeServer.Disconnect()
+                    End Try
+                End While
+            Catch ex As IOException
+            Catch ex As ObjectDisposedException
```

This will essentially hide the failure from the caller. Is the intent for the caller to just proceed with the stream and get an error when used?